@codex please fix

```
Run if [ -n "aarch64-apple-darwin" ]; then

> token-place-desktop-tauri@0.1.0 tauri
> tauri build --target aarch64-apple-darwin

        Info Looking up installed tauri packages to check mismatched versions...
     Running beforeBuildCommand `npm run build`

> token-place-desktop-tauri@0.1.0 build
> vite build

vite v5.4.21 building for production...
transforming...
✓ 34 modules transformed.
rendering chunks...
computing gzip size...
dist/index.html                  0.33 kB │ gzip:  0.24 kB
dist/assets/index-DCRzgToM.js  153.06 kB │ gzip: 48.89 kB
✓ built in 799ms
    Updating crates.io index
   Compiling token-place-desktop-tauri v0.1.0 (/Users/runner/work/token.place/token.place/desktop-tauri/src-tauri)
    Finished `release` profile [optimized] target(s) in 48.79s
       Built application at: /Users/runner/work/token.place/token.place/desktop-tauri/src-tauri/target/aarch64-apple-darwin/release/token-place-desktop-tauri
    Bundling token.place desktop.app (/Users/runner/work/token.place/token.place/desktop-tauri/src-tauri/target/aarch64-apple-darwin/release/bundle/macos/token.place desktop.app)
    Bundling token.place desktop_0.1.0_aarch64.dmg (/Users/runner/work/token.place/token.place/desktop-tauri/src-tauri/target/aarch64-apple-darwin/release/bundle/dmg/token.place desktop_0.1.0_aarch64.dmg)
     Running bundle_dmg.sh
failed to bundle project error running bundle_dmg.sh: `failed to run /Users/runner/work/token.place/token.place/desktop-tauri/src-tauri/target/aarch64-apple-darwin/release/bundle/dmg/bundle_dmg.sh`
       Error failed to bundle project error running bundle_dmg.sh: `failed to run /Users/runner/work/token.place/token.place/desktop-tauri/src-tauri/target/aarch64-apple-darwin/release/bundle/dmg/bundle_dmg.sh`
Error: Process completed with exit code 1.
```

new codex task, not a r/e/v/i/e/w task

--- a/.github/workflows/desktop-release.yml
+++ b/.github/workflows/desktop-release.yml
@@ -218,7 +218,9 @@ jobs:
 
             app_path="${app_files[0]}"
             app_name="$(basename "${app_path}" .app)"
-            dmg_path="release-artifacts/${app_name}_${{ github.ref_name || inputs.tag_name || 'manual' }}_${{ matrix.tauri_target }}.dmg"
+            ref_slug="${{ github.ref_name || inputs.tag_name || 'manual' }}"
+            ref_slug="${ref_slug//\//-}"
+            dmg_path="release-artifacts/${app_name}_${ref_slug}_${{ matrix.tauri_target }}.dmg"
 
             rm -f "${dmg_path}"
             hdiutil create -volname "${app_name}" -srcfolder "${app_path}" -ov -format UDZO "${dmg_path}"

--- a/.github/workflows/desktop-release.yml
+++ b/.github/workflows/desktop-release.yml
@@ -191,7 +191,12 @@ jobs:
       - name: Build Tauri bundles
         shell: bash
         run: |
-          if [ -n "${{ matrix.tauri_target }}" ]; then
+          if [ "${{ runner.os }}" = "macOS" ] && [ -n "${{ matrix.tauri_target }}" ]; then
+            # Tauri's built-in DMG bundler is flaky for cross-target macOS builds in CI
+            # (bundle_dmg.sh intermittently exits non-zero). Build the .app reliably, then
+            # generate the DMG in the staging step via hdiutil.
+            npm run tauri build -- --target ${{ matrix.tauri_target }} --bundles app
+          elif [ -n "${{ matrix.tauri_target }}" ]; then
             npm run tauri build -- --target ${{ matrix.tauri_target }}
           else
             npm run tauri build
@@ -203,13 +208,20 @@ jobs:
           set -euo pipefail
           mkdir -p release-artifacts
           if [ "${{ runner.os }}" = "macOS" ]; then
-            dmg_dir="src-tauri/target/${{ matrix.tauri_target }}/release/bundle/dmg"
-            dmg_files=("${dmg_dir}"/*.dmg)
-            if [ ! -e "${dmg_files[0]}" ]; then
-              echo "No DMG found in ${dmg_dir}" >&2
+            bundle_root="src-tauri/target/${{ matrix.tauri_target }}/release/bundle"
+            app_dir="${bundle_root}/macos"
+            app_files=("${app_dir}"/*.app)
+            if [ ! -e "${app_files[0]}" ]; then
+              echo "No .app bundle found in ${app_dir}" >&2
               exit 1
             fi
-            cp "${dmg_files[@]}" release-artifacts/
+
+            app_path="${app_files[0]}"
+            app_name="$(basename "${app_path}" .app)"
+            dmg_path="release-artifacts/${app_name}_${{ github.ref_name || inputs.tag_name || 'manual' }}_${{ matrix.tauri_target }}.dmg"
+
+            rm -f "${dmg_path}"
+            hdiutil create -volname "${app_name}" -srcfolder "${app_path}" -ov -format UDZO "${dmg_path}"
           elif [ "${{ runner.os }}" = "Windows" ]; then
             nsis_dir="src-tauri/target/release/bundle/nsis"
             msi_dir="src-tauri/target/release/bundle/msi"

